### PR TITLE
Config Auditor - Skip extra parameters

### LINKED ISSUE #545: Config auditor complains about extra params in multipart object
The root-level `multipart` param is where params for formidable are placed. The config auditor shouldn't care about the stuff in there.

### PR DESCRIPTION
**Pull request by mulligan121**
This makes the config auditor skip the extra parameters check for `multipart`, `bodyParser`, and `logging`. It also skips the missing parameters check for `multipart` and `bodyParser`.

Closes #545

### CODE REVIEW

File: lib/tools/checkParamObject.js [hunk 1]
```diff
@@ -7,9 +7,13 @@ module.exports = function (userParam, defaultParam, auditKey) {
   let defaultProp
   let extraOrMissingkeysBool = false
 
+  // params to skip
+  let skipExtra = ['logging', 'multipart', 'bodyParser']
+  let skipMissing = ['multipart', 'bodyParser']
```
Comment (kethinov) [inline lib/tools/checkParamObject.js]: let's declare these in config.json

i'm thinking something like this: if we add a new param like `"_roosevelt-exempt": "true"` to things in config.json that can have freeform params, the auditor will simply ignore that param and not audit it

Comment (Autre31415) [reply lib/tools/checkParamObject.js]: In order to smoothly transition to that you'll have to ensure that the auditor ignores that key, keeping in mind that the auditor and param setter share resources.

File: lib/defaults/config.json [hunk 2]
```diff
@@ -35,7 +38,9 @@
     "urlEncoded": {
       "extended": true
     },
-    "json": {}
+    "json": {},
```
Comment (Autre31415) [inline lib/defaults/config.json]: The individual bodyParser params should be exempt, not the bodyParser object itself.

Comment (kethinov) [reply lib/defaults/config.json]: ah right, good catch

Comment (kethinov) [reply lib/defaults/config.json]: what troy is saying we is we want to require the `json` and `urlEncoded` params, but we don't care what's in them

Comment (mulligan121) [reply lib/defaults/config.json]: Ah alright I can do that

**Comment by Autre31415** [comment]
`htmlMinifier.options` should be missing and extra exempt as well.

**Pull request merged**